src/policies/conditions.ts

```diff
@@ -288,13 +288,14 @@ class EvmCondition extends Condition {
       .required(),
     standardContractType: Joi.string()
       .valid(...EvmCondition.STANDARD_CONTRACT_TYPES)
-      .required(),
-    // TODO: Support custom function ABIs
-    // functionAbi: Joi.string().optional(),
+      .optional(),
+    functionAbi: Joi.object().optional(),
```

Comment:
Question Is there any way to validate a `functionAbi` dict that is provided?

Reply:
Apologies, you already filed an issue for it - https://github.com/nucypher/nucypher-ts/issues/124. Nvm

Reply:
Yes, we can use `joi` for that. It's not going to be too complicated, but I opted to address it in a separate PR since it's not critical for the release.

I'm tracking this issue here: https://github.com/nucypher/nucypher-ts/issues/124

Reply:
Makes sense 👍